### PR TITLE
Feature/5293: Configure maximum number of attempts when trying to lock SH and SP for connection creation.

### PR DESCRIPTION
**Pull request by GustavoDinizMonteiro**

### CODE REVIEW

File: networking_oneview/ml2/drivers/oneview/common.py [hunk 1]
```diff
@@ -36,6 +36,41 @@
...
+               help=('Maximum number of attempts when trying to lock Server ',
+                     'Hardware for connection creation')),
+    cfg.IntOpt('retries_to_lock_sp',
+               default=10,
+               help=('Maximum number of attempts when trying to lock Server ',
+                     'Hardware for connection creation'))
+]
+
+CONF = cfg.CONF
+CONF.register_opts(opts, group='oneview')
```
Comment (hugonicodemos) [inline networking_oneview/ml2/drivers/oneview/common.py]: this options and CONF was moved to networking_oneview/conf/oneview.py, please change there.

File: networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py [hunk 2]
```diff
@@ -58,13 +58,17 @@ def _is_physnet_in_uplinkset_mapping(self, physical_network, network_type):
         return False
 
     def check_server_hardware_availability(self, server_hardware):
-        while True:
+        max_number_of_attempts = common.CONF.oneview.retries_to_lock_sh
+
+        for i in range(max_number_of_attempts):
             if not server_hardware.get('powerLock'):
                 return True
             time.sleep(30)
```
Comment (hugonicodemos) [inline networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py]: need to create a parameter for this too.

File: networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py [hunk 3]
```diff
@@ -58,13 +58,17 @@ def _is_physnet_in_uplinkset_mapping(self, physical_network, network_type):
...
 
     def check_server_profile_availability(self, server_hardware):
-        while True:
+        max_number_of_attempts = common.CONF.oneview.retries_to_lock_sp
+
+        for i in range(max_number_of_attempts):
             if self.get_server_profile_state(server_hardware):
                 return True
             time.sleep(5)
```
Comment (hugonicodemos) [inline networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py]: ditto

**Review by hugonicodemos — COMMENTED** [review]
see the last commit for more information about the networking oneview configuration. remember to generate a new sample file with the new parameters.

File: networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py (hunk 2 above)
Comment (stenioaraujo) [reply networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py]: Maybe `retries_to_lock_sh_interval`

File: networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py (hunk 3 above)
Comment (stenioaraujo) [reply networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py]: Maybe `retries_to_lock_sp_interval`

File: networking_oneview/ml2/drivers/oneview/common.py (hunk 1 above)
Comment (hugonicodemos) [reply networking_oneview/ml2/drivers/oneview/common.py]: actually, I think those parameters are better in the DEFAULT section.

File: networking_oneview/conf/oneview.py [hunk 4]
```diff
@@ -38,9 +38,22 @@
     cfg.IntOpt('ov_refresh_interval',
                default=3600,
                help='Interval between periodic task executions in seconds.'),
-    cfg.BoolOpt('developer_mode',
```
Comment (hugonicodemos) [inline networking_oneview/conf/oneview.py]: this is not to be removed in this feature.

File: networking_oneview/conf/oneview.py [hunk 5]
```diff
@@ -38,9 +38,22 @@
...
-                default=False,
-                help='Only set this option as true if under developer mode.')
+    cfg.IntOpt('retries_to_lock_sh',
```
Comment (hugonicodemos) [inline networking_oneview/conf/oneview.py]: this 4 parameters its better in the DEFAULT section

File: networking_oneview/ml2/drivers/oneview/common.py [hunk 6]
```diff
@@ -357,18 +357,26 @@ def check_oneview_entities_availability(oneview_client, server_hardware):
     _check_server_profile_availability(oneview_client, server_hardware)
 
 
-def _check_server_profile_availability(oneview_client, server_hardware):
-    while True:
-        if _get_server_profile_state(oneview_client, server_hardware):
+def _check_server_hardware_availability(server_hardware):
```
Comment (hugonicodemos) [inline networking_oneview/ml2/drivers/oneview/common.py]: remove _

File: networking_oneview/ml2/drivers/oneview/common.py [hunk 7]
```diff
@@ -357,18 +357,26 @@ def check_oneview_entities_availability(oneview_client, server_hardware):
...
             return True
-        time.sleep(5)
+        time.sleep(interval)
+    return False
 
 
-def _check_server_hardware_availability(server_hardware):
-    while True:
-        if not server_hardware.get('powerLock'):
+def _check_server_profile_availability(oneview_client, server_hardware):
```
Comment (hugonicodemos) [inline networking_oneview/ml2/drivers/oneview/common.py]: ditto

File: networking_oneview/ml2/drivers/oneview/common.py (hunk 6 above)
Comment (ricardoas) [reply networking_oneview/ml2/drivers/oneview/common.py]: Which `_`? Is this a public method now?

File: networking_oneview/conf/oneview.py (hunk 5 above)
Comment (ricardoas) [reply networking_oneview/conf/oneview.py]: agreed... but we need some further work to enable it, then move them together with sync interval and mappings, right?
I guess we can leave them here for now and open a new issue with it!

**Review by ricardoas — CHANGES_REQUESTED** [review]
Looks good... only missing an update on the config sample with the new parameters. There is something in tox to do this...

File: networking_oneview/conf/oneview.py (hunk 4 above)
Comment (hugonicodemos) [inline networking_oneview/conf/oneview.py]: delete this only in feature/5295.

File: networking_oneview/conf/oneview.py (hunk 5 above)
Comment (hugonicodemos) [reply networking_oneview/conf/oneview.py]: create a branch (bug/5328), you can rebase with it and put those in the right section.

**Review by hugonicodemos — CHANGES_REQUESTED** [review]

**Review by hugonicodemos — APPROVED** [review]

**Review by stenioaraujo — CHANGES_REQUESTED** [review]
The code looks good, I only have one request. Could you add some tests for _check_server_hardware_availability and _check_server_profile_availability?

**Review by SauloAislan — APPROVED** [review]
Did you forget the tests?

File: networking_oneview/tests/unit/ml2/drivers/oneview/test_common.py [hunk 8]
```diff
@@ -63,3 +63,33 @@ def test_get_oneview_client_insecure_cafile(self, mock_oneview_client):
         self.credentials["ssl_certificate"] = None
         common.get_oneview_client()
         mock_oneview_client.assert_called_once_with(self.credentials)
+
+    def test_check_server_hardware_availability(
+        self, mock_oneview_client
+    ):
+        self.conf.DEFAULT.retries_to_lock_sh_interval = 1
+
+        for i in range(1, 6):
```
Comment (hugonicodemos) [inline networking_oneview/tests/unit/ml2/drivers/oneview/test_common.py]: You can simplify this, doesn't need to create a for loop. And create a positive test.

File: networking_oneview/tests/unit/ml2/drivers/oneview/test_common.py [hunk 9]
```diff
@@ -63,3 +63,33 @@ def test_get_oneview_client_insecure_cafile(self, mock_oneview_client):
...
+
+            self.assertEqual(mock_sh.get.call_count, i)
+
+    def test_check_server_profile_availability(
+        self, mock_oneview_client
+    ):
+        self.conf.DEFAULT.retries_to_lock_sp_interval = 1
+        mock_oneview_client.get_server_profile_state.return_value = None
+
+        for i in range(1, 6):
```
Comment (hugonicodemos) [inline networking_oneview/tests/unit/ml2/drivers/oneview/test_common.py]: ditto

**Review by hugonicodemos — CHANGES_REQUESTED** [review]

**Review by stenioaraujo — APPROVED** [review]

**Review by hugonicodemos — APPROVED** [review]

File: .gitignore [hunk 10]
```diff
@@ -23,6 +23,7 @@ nosetests.xml
 .testrepository
 .stestr
 .venv
+coverage.xml
```
Comment (ricardoas) [inline .gitignore:26]: @hugonicodemos does this mess with our coverage report????

**Review by ricardoas — APPROVED** [review]

**Pull request merged**